app/containers/SignInPage/reducer.js

```diff
@@ -0,0 +1,24 @@
+import { fromJS } from 'immutable';
+
+import {
+  REQUEST_SIGN_IN_SUCCESS,
+  REQUEST_SIGN_IN_FAILED,
+} from './constants';
+
+export const initialState = fromJS({
+  token: "",
```

Comment:
The token shouldn't be part of the SignInPage state because you need it in all the app.